ui/src/components/TrafficPage.tsx

```diff
@@ -100,16 +101,30 @@ export const TrafficPage: React.FC<TrafficPageProps> = ({setAnalyzeStatus, onTLS
 
     const listEntry = useRef(null);
 
-    const openWebSocket = (query) => {
-        setFocusedEntryId(null);
-        setEntries([]);
-        setEntriesBuffer([]);
+    const openWebSocket = (query: string, dontClear: boolean) => {
+        if (!dontClear) {
+            setFocusedEntryId(null);
+            setEntries([]);
+            setEntriesBuffer([]);
+        } else {
+            setEntriesBuffer(entries);
+        }
         ws.current = new WebSocket(MizuWebsocketURL);
         ws.current.onopen = () => {
-            ws.current.send(query)
             setConnection(ConnectionStatus.Connected);
+            ws.current.send(query);
+        }
+        ws.current.onclose = () => {
+            setConnection(ConnectionStatus.Closed);
+        }
+        ws.current.onerror = (event) => {
+            console.error("WebSocket error:", event);
+            if (query) {
+                openWebSocket(`(${query}) and leftOff(${leftOff})`, true);
```

Comment:
do we want to support leftOff that will be added by user? 
because then you can get 
query that has "http and leftOff(-1) and leftOff(600)" which is weird.
but maybe I misunderstand and leftOff is internal parameter that should not be specified by user

Reply:
@gadotroee it's aimed to be used internally. There is no accurate information in the UI for user to provide a correct value also. They can do inspect element to get the ids though.